Setup for neptune tools and scripts

mihails-strasuns-sociomantic opened the pull request:
Provides one small library module for working with SemVer like
version strings and necessary standard Sociomantic project boilerplate.

gavin-norman-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+
+    Utility to simplify working with semantic versioning entities, including
+    parsing and comparison.
+
```
Needs copyright and licensing info.

gavin-norman-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+     **/
+    string toString ( ) const pure
+    {
+        return this.representation;
```
`this.representation` will be invalid, if an instance is constructed manually (i.e. not via `fromString()`).

gavin-norman-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+        assert(ver.representation == "v30.20.10");
+
+        ver = Version.fromString("gibberish");
+        assert(ver.representation == "gibberish");
```
Also assert that the numbers are `int.init`?

leandro-lucarella-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+        import std.conv;
+        import std.regex;
+
+        static verRegex = regex(r"^v(\d+)\.(\d+)\.(\d+)$", "g");
```
Even when we use always `vX.Y.Z`maybe it would be good to be more flexible and accept strings without the leading `v` too (`regex(r"^v?(\d+)\.(\d+)\.(\d+)$", "g");`). Maybe it can accept a `bool` for strict mode.

leandro-lucarella-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+    int minor;
+
+    /// patch version number
+    int patch;
```
I think this is missing some parts of semver: [pre-release](http://semver.org/#spec-item-9) and [build metadata](http://semver.org/#spec-item-10). We'll need it for `+breaking` for example.

mihails-strasuns-sociomantic replied:
Let's create separate issue for that. I have created this PR to quickly add something for basic project layout and travis testing :P

leandro-lucarella-sociomantic replied:
Sure, but please create the issue :)

mihails-strasuns-sociomantic replied:
https://github.com/sociomantic-tsunami/neptune/issues/11

leandro-lucarella-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+
+        result.major = to!int(hit[1]);
+        result.minor = to!int(hit[2]);
+        result.patch = to!int(hit[3]);
```
Check `int`s to be non-negative (is specified by semver).

leandro-lucarella-sociomantic commented on src/semver/Version.d:
```diff
@@ -0,0 +1,139 @@
+        assert(ver.patch == 10);
+        assert(ver.representation == "v30.20.10");
+
+        ver = Version.fromString("gibberish");
```
So this is not validated at all? Shouldn't it throw if the string is not in the expected format?

mihails-strasuns-sociomantic commented:
Updated:

- removed support for non-compatible version strings (it was a relict from old days when I had to support all sorts of weird tags in our repos)
- `v` prefix is optional
- .travis.yml added
- disabled default ctor so that version can be only created by calling `parse`

mihails-strasuns-sociomantic commented:
Updated:
- instead of disabling ctor, explicitly define one that requires setting all 3 fields (that will disable all automatic ones)

leandro-lucarella-sociomantic commented on Makefile:
```diff
@@ -0,0 +1,7 @@
+include submodules/makd/Makd.mak
+
+# Default goal for building this directory
+.DEFAULT_GOAL := test
```
Why test?
